### PR TITLE
feat(multisig): add multisig service

### PR DESCRIPTION
**Pull request by sdaveas**

### CODE REVIEW

File: multisig.proto [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    string key_uid = 1;
+    bytes msg_to_sign = 2;
+}
+
+message SignResponse { 
+    oneof sign_response {
+        bytes signature = 1;
+        string error = 2; // reply with an error message if sign fails
+    }
+}
```
Comment (milapsheth) [inline multisig.proto:34]: Newline?

**Review by milapsheth — APPROVED** [review]

File: multisig.proto [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+message KeygenRequest {
+    string key_uid = 1;
+    string party_uid = 2;
+}
+
+message KeygenResponse { 
+    oneof keygen_response {
+        bytes pub_key = 1;
```
Comment (ggutoski) [inline multisig.proto:19]:
```suggestion
        bytes pub_key = 1; // SEC1-encoded compressed curve point
```

File: multisig.proto [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+message SignRequest {
+    string key_uid = 1;
+    bytes msg_to_sign = 2;
+}
+
+message SignResponse { 
+    oneof sign_response {
+        bytes signature = 1;
```
Comment (ggutoski) [inline multisig.proto:31]:
```suggestion
        bytes signature = 1; // ASN.1 DER-encoded ECDSA signature
```

File: multisig.proto [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+message KeygenResponse { 
+    oneof keygen_response {
+        bytes pub_key = 1;
+        string error = 2; // reply with an error message if keygen fails
+    }
+}
+
+message SignRequest {
+    string key_uid = 1;
+    bytes msg_to_sign = 2;
```
Comment (ggutoski) [inline multisig.proto:26]:
```suggestion
    bytes msg_to_sign = 2; // 32-byte pre-hashed message digest
```

File: multisig.proto [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+package tofnd;
+
+service Multisig {
+    rpc Keygen(KeygenRequest) returns (KeygenResponse);
+    rpc Sign(SignRequest) returns (SignResponse);
+}
+
+message KeygenRequest {
+    string key_uid = 1;
+    string party_uid = 2;
```
Comment (ggutoski) [inline multisig.proto:14]: Do we need `party_uid`?

**Review by ggutoski — APPROVED** [review]

File: multisig.proto (hunk 5 above)
Comment (ggutoski) [reply multisig.proto:14]:
```suggestion
    string party_uid = 2; // used only for logging
```

**Review by milapsheth — APPROVED** [review]

**Pull request merged**